pandas/core/algorithms.py

```diff
@@ -533,7 +533,7 @@ def factorize(
     sort: bool = False,
     na_sentinel: Optional[int] = -1,
     size_hint: Optional[int] = None,
-) -> Tuple[np.ndarray, Union[np.ndarray, ABCIndex]]:
```

Comment:
not sure if it matters for annotations, but `isinstance(obj, ABCIndex)` is True for only Index, _not_ subclasses.  Does that make a difference here?

Reply:
Looks like the return should included subclasses. I added:

```
from pandas import Index
assert isinstance(uniques, np.ndarray) or type(uniques) is Index, type(uniques)
```

above the return of this function and from conftest:412 got 

> AssertionError: <class 'pandas.core.indexes.datetimes.DatetimeIndex'>